godtools/Views/TractElements/BaseTractElement.swift

```diff
@@ -12,13 +12,56 @@ import SWXMLHash
 
 class BaseTractElement: UIView {
     
+    // MARK: Positions constants
+    
     static let xMargin: CGFloat = 8.0
     static let yMargin: CGFloat = 8.0
     static let xPadding: CGFloat = 0.0
     static let yPadding: CGFloat = 0.0
     static let screenWidth = UIScreen.main.bounds.size.width
     static let textContentWidth = UIScreen.main.bounds.size.width - BaseTractElement.xMargin * CGFloat(2)
     
+    // MARK: - Positions and Sizes
+    
+    var yStartPosition: CGFloat = 0.0
+    var maxHeight: CGFloat = 0.0
+    var height: CGFloat = 0.0
+    var width: CGFloat {
+        if (self.parent != nil) {
+            return self.parent!.width
+        } else {
+            return BaseTractElement.screenWidth
+        }
+    }
+    
+    func yEndPosition() -> CGFloat {
+        return self.yStartPosition + self.height
+    }
+    
+    func getMaxHeight() -> CGFloat {
+        if self.maxHeight > 0.0 {
+            return self.maxHeight
+        } else if (self.parent != nil) {
+            return (self.parent?.getMaxHeight())!
+        } else {
+            return 0.0
+        }
+    }
+    
+    func textYPadding() -> CGFloat {
+        return BaseTractElement.yPadding
+    }
+    
+    // MARK: Main properties
+    
+    private var _mainView: TractRoot?
+    var root: TractRoot? {
+        get {
+            let parentRoot = self.parent?.root
+            return self._mainView != nil ? self._mainView : parentRoot!
```

Comment:
Actually I'm changing this on another PR https://github.com/CruGlobal/godtools-swift/pull/23/files#diff-051201f9a19e813e283035484372d8afR60 for now I will merge it because this will change in the other PR